oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/tree/TreeStore.java

```diff
@@ -166,4 +171,28 @@ public Store getStore() {
         return store;
     }
 
+    @Override
+    public String getStorePath() {
+        return directory.getAbsolutePath();
+    }
+
+    @Override
+    public long getEntryCount() {
+        return entryCount;
+    }
+
+    public void setEntryCount(long entryCount) {
+        this.entryCount = entryCount;
+    }
+
+    @Override
+    public String getIndexStoreType() {
+        return STORE_TYPE;
+    }
+
+    @Override
+    public boolean isIncremental() {
+        return true;
```

Comment:
This should be false. 

The TreeStore should be the base implementation - that builds the a tree representation of the whole repository, and then we can have something like an IncrementalTreeStore, which builds the diff b/w 2 checkpoints to be appended to the base TreeStore.

Or maybe simply use the diff by IncrementalFlatFileStore and merge that in the base TreeStore.

Refer - 

IndexStore -> FlatFileStore -> IncrementalFlatFileStore

Reply:
To update the tree store incrementally, we can use log-structured merges, using Session.checkpoint() / Session.flush(). In theory, this is much more efficient than the IncrementalFlatFileStore, if done correctly. Specially if there are few changes. This is already implemented in the tree store, but needs to be "wired" correctly. 

But I think I will not yet do that just yet, but instead concentrate on getting the basics working correctly (right now, e.g. the cache size is in number of entries instead of memory used... this can lead to OOME if we have very unevenly sizes pages). Also, I need to port some unit tests.

So I'll set it to "false" currently.